datastore/google/cloud/datastore/_http.py

```diff
@@ -61,6 +62,9 @@
 
 _DISABLE_GRPC = os.getenv(DISABLE_GRPC, False)
 _USE_GRPC = _HAVE_GRPC and not _DISABLE_GRPC
+_DATASTORE_DIST = get_distribution('google-cloud-datastore')
+_CLIENT_INFO = connection_module.CLIENT_INFO_TEMPLATE.format(
+    _DATASTORE_DIST.version)
```

Comment:
You've succinctly summarized the exact conversation that @lukesneeringer and I had yesterday :grinning: 

The "right" thing to do would be to say something like `core-version/X.Y.Z FOO-version/a.b.c` but that decision is out of our hands. So we are using `gccl/a.b.c` (i.e. `FOO`'s version) and relying on the actual backend service to determine which subpackage we meant.